packages/nimble-components/src/rich-text/specs/README.md

```diff
@@ -107,6 +111,114 @@ Example usage of the `nimble-rich-text-editor` in the application layer is as fo
+   a new configuration component can be created and added as a child to the rich text editor. Below is an example of how
+   these elements would be used:
+
+```html
+<nimble-rich-text-editor>
+    <nimble-rich-text-mention-user>
+        <nimble-mapping-mention
+            key="user-id-1"
+            text="John Doe"
+        ></nimble-mapping-mention>
+        <nimble-mapping-mention
+            key="user-id-2"
+            text="Alice Smith"
+        ></nimble-mapping-mention>
+        <nimble-mapping-mention
+            key="user-id-3"
+            text="Bob Jones"
+        ></nimble-mapping-mention>
+    </nimble-rich-text-mention-user>
+
+    <nimble-rich-text-mention-issue>
+        <nimble-mapping-mention
+            key="issue-id-1"
+            text="Spec for rich text editor"
+        ></nimble-mapping-mention>
+        <nimble-mapping-mention
+            key="issue-id-2"
+            text="Mention support in rich text components"
+        ></nimble-mapping-mention>
+        <nimble-mapping-mention
+            key="issue-id-3"
+            text="Issue in pasting a link"
+        ></nimble-mapping-mention>
+    </nimble-rich-text-mention-issue>
+</nimble-rich-text-editor>
+```
+
+_Alternatives_:
+
+Slot based approach: All user information will be communicated through `nimble-list-option` elements. These elements will
+contain the user's ID in their `value` attribute and the user's name in their text content. When the **"@"** symbol is
+inserted into the editor, these elements will be presented as a dropdown list.
+
+Pros:
+
+1. When the client does not specify a `value` attribute, the options are sent without a user's ID, and in such cases,
+   the value attribute is derived from the content of the option.
+
+Cons:
+
+1. It is impossible to have multiple type of mentions when `nimble-list-option` is used in the default slot. If we use
+   slot names to differentiate between various type of mentions, the rich text editor needs to be aware of all the type
+   of mentions like user slot name, issue slot name etc, through an attribute value.
+2. The slot-based approach is not configurable, meaning that if a user wishes to customize the entire user mention with
+   some arbitrary value, the only viable option is to include it within the options. In contrast, the configuration-based
+   approach readily allows for the inclusion of arbitrary values in the parent container element, making customization more
+   straightforward.
+3. The UI of the list option is completely dependent on the client application.
+4. All the keyboard and mouse interactions within the dropdown list should be handled internally within the editor as `nimble-list-option`
+   does not support any.
+
+#### Client Usage Guidance on Filtered Users:
+
+Initially, the list of users can be sorted in alphabetical order based on their usernames. Then, within the `nimble-mapping-mention`,
```

Comment:
Yes, it is not required to include the entire user list initially. It can be updated dynamically only after the `@` character is entered into the editor by listening to the event for that purpose. I have updated the spec accordingly to align with your point.